src/components/DepartmentListComponent.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FC } from 'react';
+
+import { useGetDepartmentListQuery } from 'services/api';
+import PopUp from './PopUp';
+import { DeleteIcon, EditIcon } from 'assets/icons';
+
+const DepartmentListComponent: FC = () => {
+
+    const { data ,error, isLoading} = useGetDepartmentListQuery();
+
+    if (isLoading){
+        return<div>Loading</div>;
+    }
+
+    if (error){
+        return<PopUp description={'Cannot load Department List'} 
+        margin={'absolute inset-x-0 bottom-16 h-16 w-[15%] min-w-[450px] border-rose-600 bg-red-50'}></PopUp>;
+    }
+    return (
+        <table className="mx-auto w-full min-w-[450px] table-auto">
+            <thead>
+                <tr className=" mt-8 flex  h-16 w-[96%]  gap-10 bg-aliceBlue p-5  text-center shadow-xl">
+                    <th className="w-1/5 font-normal">Department Id</th>
+                    <th className="w-1/5 font-normal">Department Name</th>
+                    <th className="w-1/5 font-normal">Website</th>
+                    <th className="w-1/5 font-normal">Location</th>
+                    <th className="w-1/5 font-normal">Actions</th>
+                </tr>
+            </thead>
+            <tbody>
+                {data?.map(department => {
```

Comment:
pass the data after mapping from the parent container.

deptData?.map((dept) => (
    <DepartmentListComponent 
         data={dept}
    />


here it would be dept.name, dept.id etc directly used in the place of variables without the map function.